back-end/user_app/views.py

```diff
@@ -0,0 +1,55 @@
+from django.shortcuts import render
+from .models import App_user
+from django.contrib.auth import authenticate
+from rest_framework.authtoken.models import Token
+from rest_framework.views import APIView
+from rest_framework.response import Response
+from rest_framework.permissions import IsAuthenticated
+from rest_framework.authentication import TokenAuthentication
+from rest_framework.status import (
+    HTTP_200_OK,
+    HTTP_201_CREATED,
+    HTTP_204_NO_CONTENT,
+    HTTP_400_BAD_REQUEST,
+)
+from django.http import HttpResponse
+
+# Create your views here.
+
+
+class Log_in(APIView):
+    def post(self, request):
+        request.data["username"] = request.data["email"]
+        user = authenticate(**request.data)
+        if user:
+            token, created = Token.objects.get_or_create(user=user)
+            return Response({"user": {"email": user.email}, "token": token.key})
+        else:
+            return Response("Sorry, something went wrong!", status=HTTP_400_BAD_REQUEST)
+
+
```

Comment:
This codebase seems oddly familiar 🤣

Reply:
Hahaha I know I didn't want to just pull this out of your code but this is the exact user functionality I was looking for so I just recreated it and didn't look back. Now theres definitely no excuse not to have a fully functional and well styled website 🤣